src/main/java/com/rebuy/consul/listeners/ContextClosedEventListener.java

```diff
@@ -0,0 +1,21 @@
+package com.rebuy.consul.listeners;
+
+import com.rebuy.consul.ConsulService;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextClosedEvent;
+
+public class ContextClosedEventListener implements ApplicationListener<ContextClosedEvent>
```

Comment:
These class names are very generic. May this cause ambiguity on using this in other projects?